src/api-v1.js

```diff
@@ -580,6 +580,213 @@ api.declare({
 
 api.declare({
   method: 'put',
+  route: '/ami-set/:id',
+  name: 'createAmiSet',
+  input: 'create-ami-set-request.json#',
+  deferAuth: true,
+  scopes: [['aws-provisioner:manage-ami-set:<amiSetId>']],
+  title: 'Create new AMI Set',
+  stability:  base.API.stability.stable,
+  description: [
+    'Create an AMI Set. An AMI Set is a collection of AMIs with a single name.',
+  ].join('\n'),
+}, async function (req, res) {
+  let input = req.body;
+  let id = req.params.id;
+
+  input.lastModified = new Date();
+
+  // Authenticate request with parameterized scope
+  if (!req.satisfies({id: id})) {
+    return;
+  }
+
+  // Create amiSet
+  let amiSet;
+  try {
+    amiSet = await this.AmiSet.create({
+      id: id,
+      amis: input.amis,
+      lastModified: new Date(),
+    });
+  } catch (err) {
+    // We only catch EntityAlreadyExists errors
+    if (!err || err.code !== 'EntityAlreadyExists') {
+      throw err;
+    }
+    amiSet = await this.AmiSet.load({id});
+
+    // Check if it matches the existing amiSet
+    let match = [
+      'id',
+      'amis',
+    ].every((key) => {
+      return _.isEqual(amiSet[key], input[key]);
+    });
+
+    // If we don't have a match we return 409, otherwise we continue as this is
+    // is an idempotent operation.
+    if (!match) {
+      res.status(409).json({
+        error: 'AMI Set already exists with different definition',
+      });
+      return;
+    }
+  }
+  res.reply({outcome: 'success'});
+  return;
+
+});
+
+api.declare({
+  method: 'get',
+  route: '/ami-set/:id',
+  name: 'getAmiSet',
+  output: 'get-ami-set-response.json#',
+  deferAuth: true,
+  scopes: [
+    ['aws-provisioner:view-ami-set:<amiSetId>'],
```

Comment:
Here, `getAmiSet` requires `aws-provisioner:view-ami-set:<id>`, but `listAmiSets` requires no scopes.  It's tricky to enforce scopes in a list method, and I don't see a great reason to hide AMI sets from public view, so I think we should just allow anyone to call `getAmiSet`, too (meaning, remove the `scopes:` here and the `req.satisifies` below).